src/Parser/BrewmasterMonk/Modules/Features/Energy.js

```diff
@@ -0,0 +1,125 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import { SPELLS_WITH_ENERGY_COST, SPELLS_WITH_HASTE } from '../../Constants';
+
+const debug = false;
+const BASE_REGEN = 10;
+export const MAX_ENERGY = 100;
+
+class Energy extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  energy = [];
```

Comment:
Those are a lot of public properties, this is generally a good indicator that some refactoring is needed.